plugins/modules/source_control/git_config.py

```diff
@@ -245,7 +245,13 @@ def main():
         else:
             new_value_quoted = shlex_quote(new_value)
             cmd = ' '.join(args + [new_value_quoted])
-        (rc, out, err) = module.run_command(cmd, cwd=dir)
+        try:  # extra parameter from ansible-base 2.10.4 onwards
+            (rc, out, err) = module.run_command(cmd, cwd=dir, ignore_invalid_cwd=False)
+        except TypeError:
+            # @TODO remove try/except when community.general drop support for 2.10.x
+            module.warn('If "repo" is an invalid directory, "git_config" will not raise an error and will run '
```

Comment:
Now every user of this module who has ansible-base 2.10.3 or earlier, or Ansible 2.9, will always get this warning. How about only printing it when the directory does not exist? Or simply fail here when the directory does not exist.

Reply:
I like the idea of warning people, as annoying as that might be. But will not press the point, so going with the _fail_ option.